lib/cocoapods/installer/analyzer.rb

```diff
@@ -498,6 +498,7 @@ def generate_pod_targets(specs_by_target)
           pod_targets.each do |target|
             dependencies = transitive_dependencies_for_specs(target.specs.reject(&:test_specification?), target.platform, all_specs).group_by(&:root)
             test_dependencies = transitive_dependencies_for_specs(target.specs.select(&:test_specification?), target.platform, all_specs).group_by(&:root)
+            test_dependencies.delete_if { |k| dependencies.key? k }
```

Comment:
This removes duplicated pod target dependencies that exist as normal dependencies but also as test dependencies.